pkg/api/v1alpha1/tinkerbellmachineconfig_types_test.go

```diff
@@ -0,0 +1,100 @@
+package v1alpha1
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestTinkerbellMachineConfigValidateSucceed(t *testing.T) {
+	machineConfig := createTinkerbellMachineConfig()
+
+	g := NewWithT(t)
+	g.Expect(machineConfig.Validate()).To(Succeed())
+}
+
+func TestTinkerbellMachineConfigValidateFail(t *testing.T) {
+	tests := []struct {
+		name          string
+		machineConfig *TinkerbellMachineConfig
+		expectedErr   string
+	}{
+		{
+			name: "Invalid object meta",
+			machineConfig: createTinkerbellMachineConfigWithOptions(func(mc *TinkerbellMachineConfig) {
+				mc.ObjectMeta.Name = ""
+			}),
+			expectedErr: "TinkerbellMachineConfig: missing name",
+		},
+		{
+			name: "Empty hardware selector",
+			machineConfig: createTinkerbellMachineConfigWithOptions(func(mc *TinkerbellMachineConfig) {
+				mc.Spec.HardwareSelector = nil
+			}),
+			expectedErr: "TinkerbellMachineConfig: missing spec.hardwareSelector",
+		},
+		{
+			name: "Multiple hardware selectors",
+			machineConfig: createTinkerbellMachineConfigWithOptions(func(mc *TinkerbellMachineConfig) {
+				mc.Spec.HardwareSelector["type2"] = "cp2"
+			}),
+			expectedErr: "TinkerbellMachineConfig: spec.hardwareSelector must contain only 1 key-value pair",
+		},
+		{
+			name: "Empty OS family",
+			machineConfig: createTinkerbellMachineConfigWithOptions(func(mc *TinkerbellMachineConfig) {
+				mc.Spec.OSFamily = ""
+			}),
+			expectedErr: "TinkerbellMachineConfig: missing spec.osFamily",
+		},
+		{
+			name: "Invalid OS family",
+			machineConfig: createTinkerbellMachineConfigWithOptions(func(mc *TinkerbellMachineConfig) {
+				mc.Spec.OSFamily = "invalid OS"
+			}),
+			expectedErr: "unsupported spec.osFamily (invalid OS); Please use one of the following",
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			g := NewWithT(t)
+			g.Expect(tc.machineConfig.Validate()).To(MatchError(ContainSubstring(tc.expectedErr)))
+		})
+	}
+}
+
+func createTinkerbellMachineConfigWithOptions(options ...func(config *TinkerbellMachineConfig)) *TinkerbellMachineConfig {
```

Comment:
This function is superfluous because options can be passed to the `createTinkerbellMachineConfig()`.